src/rebar_dialyzer.erl

```diff
@@ -0,0 +1,235 @@
+         'check-plt'/2,
+         'delete-plt'/2
+        ]).
+
+%% for internal use only
+-export([info/2]).
+
+-include("rebar.hrl").
+
+%% ===================================================================
+%% Public API
+%% ===================================================================
+
+%% TODO:
+%%
+%%  * How can we configure the filtering of warnings without exact line
+%%    numbers as a way to avoid updating the config whenever the file
+%%    in question is modified (and a warning's line number changes)?
+%%    Maybe use '_' for line number in dialyzer_ignored_warnings or a
+%%    separate dialyzer_ignored_warnings_noline (name to be determined)
+%%    list, and subtract that from the dialyzer:run/2 result.
+
+dialyze(Config, AppFile) ->
+    {NewConfig, Plt} = plt(Config, AppFile),
+    ok = check_plt_existence(Plt),
+
+    Opts = [
+            {analysis_type, succ_typings},
+            {init_plt, Plt},
+            {files_rec, ["ebin"]},
+            {warnings, warnings(Config)}
+           ],
+    ?DEBUG("dialyze opts:~n~p~n", [Opts]),
+    RawWs = dialyzer:run(Opts),
+    case filter_warnings(RawWs, Config) of
+        [] ->
+            {ok, NewConfig};
+        Ws ->
+            print_warnings(Ws),
+            ?FAIL
+    end.
+
+'build-plt'(Config, AppFile) ->
+    {Config1, AppDirs} = app_dirs(Config, AppFile),
+    {NewConfig, Plt} = plt(Config1, AppFile),
+    Opts = [
+            {analysis_type, plt_build},
+            {output_plt, Plt},
+            {files_rec, AppDirs}
+           ],
+    ?DEBUG("build-plt opts:~n~p~n", [Opts]),
+    case dialyzer:run(Opts) of
+        [] ->
+            {ok, NewConfig};
+        Ws ->
+            %% As plt_build may raise warnings but still successfully
+            %% create the PLT, we cannot interpret this as failure,
+            %% and therefore all we can do is report warnings.
+            print_warnings(Ws)
+    end.
+
+'check-plt'(Config, AppFile) ->
+    {NewConfig, Plt} = plt(Config, AppFile),
+    ok = check_plt_existence(Plt),
+
+    Opts = [
+            {analysis_type, plt_check},
+            {init_plt, Plt}
+           ],
+    ?DEBUG("build-plt opts:~n~p~n", [Opts]),
+    case dialyzer:run(Opts) of
+        [] ->
+            {ok, NewConfig};
+        Ws ->
+            print_warnings(Ws),
+            ?FAIL
+    end.
+
+'delete-plt'(Config, AppFile) ->
+    {NewConfig, Plt} = plt(Config, AppFile),
+    ?DEBUG("Delete PLT '~s'~n", [Plt]),
+    ok = rebar_file_utils:delete_each([Plt]),
+    {ok, NewConfig}.
+
+%% ===================================================================
+%% Internal functions
+%% ===================================================================
+
+info(help, dialyze) ->
+    info_help("Analyze the code for discrepancies");
+info(help, 'build-plt') ->
+    info_help("Build project-specific PLT");
+info(help, 'check-plt') ->
+    info_help("Check the PLT for consistency and rebuild it if it"
+              " is not up-to-date");
```

Comment:
I'd have to reconstruct the test case, but the results were that it isn't always clear what was added/removed/updated, because the available data isn't enough to get the difference. Therefore, I've discarded the idea of a custom update-plt function. Usually `check-plt`'s update of the PLT should suffice, and if it fails a clean `build-plt` is the right choice.